File: Data Structures & Algorithms/remove-element/submission-1.cpp
class Solution {
public:
    int removeElement(vector<int>& nums, int val) {
        int n = nums.size(), k = n - 1;
        int i = 0, cnt = 0;
        while (i <= k) {
            if (nums[i] == val) {
                nums[i] ^= nums[k];
                nums[k] ^= nums[i];
                nums[i] ^= nums[k];
                k--;
                cnt++;
            } else {
                i++;
            }
        }
        return n - cnt;
    }
};

File: Data Structures & Algorithms/longest-consecutive-sequence/submission-1.java
class Solution {
    public int longestConsecutive(int[] nums) {
        Set<Integer> seen = new HashSet<>();
        int longest = 0, cnt = 0;
        for (int n : nums){
            seen.add(n);
        }
        for (int n : seen){
            if (!seen.contains(n - 1)){
                cnt = 0;
                while (seen.contains(n)){
                    cnt += 1;
                    n += 1;
                }
                longest = Math.max(longest, cnt);
            }
        }
        return longest;
    }
}
